Detectors/TRD/workflow/src/TRDTrackingWorkflow.cxx

```diff
@@ -43,11 +44,13 @@ framework::WorkflowSpec getTRDTrackingWorkflow(bool disableRootInp, bool disable
       specs.emplace_back(o2::tpc::getTPCTrackReaderSpec(useMC));
     }
     specs.emplace_back(o2::trd::getTRDTrackletReaderSpec(useMC, false));
+    if (trigRecFilterActive) {
+      specs.emplace_back(o2::its::getIRFrameReaderSpec());
+    }
```

Comment:
I tried that yesterday, but it is not straightforward for the TRD. The problem is that we have two tracklet types: Our tracklets which come from the FEE `Tracklet64` and the space points which are calculated from those tracklets with calibration applied called `CalibratedTracklet`. These come from the tracklet transformer.
The InputHelper tries to load everything for the TRD if TRD clusters are asked for. Basically I need another argument for ``InputHelper::addInputSpecs()` to distinguish between the two cases (if the calibrated tracklets should be loaded as well or not). Or I separate the tracklet transformer workflow from the tracking workflow. Then I can use the InputHelper for the tracking, but not for the tracklet transformer.

Reply:
Not sure I understand the problem: you are anyway reading the Tracklet64, which could be done via InputHelper, and then you can couple this reading to transormer, like it is done for the TPC shared cluster creator: https://github.com/AliceO2Group/AliceO2/blob/ae21d9f42c0c2469d6e5595af955f6ba1dd4c15c/Detectors/GlobalTrackingWorkflow/helpers/src/InputHelper.cxx#L67
So, *if* you read the tracklets from the file, the InputHelper would also provide the CalibratedTracklet.
What is more tricky is having the "strict" and "relaxed" workflows in the same piped workflow, receiving the Tracklet64 from the upstream (e.g. from CTF-reader or raw-reader/DD):  you should either add to getTRDTrackingWorkflow a flag saying that the transformer should be attached (true in "strict" and false in "relaxed" instances) or to attach the transformer to the raw->tracklets and entropy decoders.

I guess when the Tracklet64 input is needed, in general, also the CalibratedTracklet will be needed? To me this is similar to TPC shared clusters container, 

````
{strict phase} [ read tracklets, ]
````

Reply:
The tracklet transformer is always needed (for the TRD tracking, for the TRD calibration and also for the TPC interpolation for the refit). The InputHelper always loads both the Tracklet64 and the CalibratedTracklets. Since the latter are not available before the tracklet transformer has run I don't use the InputHelper for the tracklet transformer. But now I have decoupled the tracklet transformer and the global tracking so that I can use the InputHelper for the global tracking at least.
So TRD tracking for the sync. reco can now be run with:
`o2-trd-tracklet-transformer -b --filter-trigrec | o2-trd-global-tracking -b --tracking-sources ITS-TPC --run --configKeyValues "GPU_rec_trd.nSigmaTerrITSTPC=4;"`

Reply:
I had some junk copy/pasted in prev. message. OK, this will work also, but I thought you can calculate the calibTracklets on the fly, reading only Tracklet64 from the file.

Reply:
Yes, I can also calculate calibTracklets on the fly. This is what happens with the above command. Read Tracklet64, calculate calibTracklets and pass them to the TRD tracker. 
To handle everything with the InputHelper would mean I need to distinguish in `DataRequest::requestTRDTracklets()` whether the calibTracklets are already there or not. And this I did not manage without breaking the logic in the RecoContainer.